app/code/Magento/Catalog/Block/Adminhtml/Category/Tab/Product.php

```diff
@@ -29,20 +32,37 @@ class Product extends \Magento\Backend\Block\Widget\Grid\Extended
      */
     protected $_productFactory;
 
+    /**
+     * @var Status
+     */
+    protected $_status;
+
+    /**
+     * @var Visibility
+     */
+    protected $_visibility;
```

Comment:
New class variables should be private, and should not be prefixed with an underscore to indicate visibility